java-frontend/src/main/java/org/sonar/java/bytecode/se/BytecodeSECheck.java

```diff
@@ -62,7 +97,133 @@ private static boolean isInvokeOnObjectRef(Instruction inst) {
+        case DSUB:
+        case FSUB:
+        case ISUB:
+        case LSUB:
+          return handlePlusMinus(previousState, currentState);
+        case DMUL:
+        case FMUL:
+        case IMUL:
+        case LMUL:
+          return handleMultiply(previousState, currentState);
+        case DDIV:
+        case FDIV:
+        case IDIV:
+        case LDIV:
+        case DREM:
+        case FREM:
+        case IREM:
+        case LREM:
+          return handleDivisionRemainder(previousState, currentState);
+        case ISHL:
+        case LSHL:
+        case ISHR:
+        case LSHR:
+        case IUSHR:
+        case LUSHR:
+          return handleShift(previousState, currentState);
+        case INEG:
+        case LNEG:
+        case FNEG:
+        case DNEG:
+          return handleNegation(previousState, currentState);
+        default:
+          break;
+      }
+      return currentState;
+    }
+
+    private static ProgramState handlePlusMinus(ProgramState previousState, ProgramState currentState) {
+      List<SymbolicValue> operands = previousState.peekValues(2);
+      SymbolicValue result = currentState.peekValue();
+      SymbolicValue op1 = operands.get(0);
+      SymbolicValue op2 = operands.get(1);
+      boolean op1Zero = isZero(currentState, op1);
+      boolean op2Zero = isZero(currentState, op2);
+      if (op1Zero && op2Zero) {
+        // Reuse one of the zero
+        return currentState.unstackValue(1).state.stackValue(op1);
+      } else if (op1Zero) {
+        // Reuse other operand
+        return currentState.unstackValue(1).state.stackValue(op2);
+      } else if (op2Zero) {
+        // Reuse other operand
+        return currentState.unstackValue(1).state.stackValue(op1);
+      }
+      // we know nothing about zero-ness
+      return currentState.removeConstraintsOnDomain(result, BooleanConstraint.class);
+    }
+
+    private static ProgramState handleMultiply(ProgramState previousState, ProgramState currentState) {
+      List<SymbolicValue> operands = previousState.peekValues(2);
+      SymbolicValue result = currentState.peekValue();
+      SymbolicValue op1 = operands.get(0);
+      SymbolicValue op2 = operands.get(1);
+      boolean op1Zero = isZero(currentState, op1);
+      if (op1Zero || isZero(currentState, op2)) {
+        // Reuse zero
+        return currentState.unstackValue(1).state.stackValue(op1Zero ? op1 : op2);
+      }
+      return currentState.removeConstraintsOnDomain(result, BooleanConstraint.class).addConstraint(result, ZeroConstraint.NON_ZERO);
+    }
+
+    private static ProgramState handleDivisionRemainder(ProgramState previousState, ProgramState currentState) {
+      List<SymbolicValue> operands = previousState.peekValues(2);
+      SymbolicValue result = currentState.peekValue();
+      SymbolicValue op1 = operands.get(0);
+      SymbolicValue op2 = operands.get(1);
+      if (isZero(currentState, op2)) {
+        // Division by zero
+        // TODO exceptional yield ?
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1135)